FxAClient/Frontend/SignIn/FxASignInWebViewController.swift

```diff
@@ -0,0 +1,179 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import Foundation
+import UIKit
+import WebKit
+
+protocol FxASignInWebViewControllerDelegate {
+    func didCancel() -> Void
```

Comment:
It is common to give delegate methods more descriptive names like:

```
func signInWebViewControllerDidCancel(signInWebViewController: FxASignInWebViewController)
func signInWebViewControllerDidLoad(signInWebViewController: FxASignInWebViewController)
func signInWebViewControllerDidSignIn(signInWebViewController: FxASignInWebViewController, withAccount account: JSON)
```
I know this is all very verbose but it is how most things in Cocoa work. One of the advantages of passing self to the delegate methods is that the class that implements the delegate then does not have to keep a reference to the viewcontroller calling the delegate. May not be relevant here but I think it is good to follow Cocoa/UIKit style.

It also prevents name conflicts. You can have a `signInWebViewController(:, didFailWithError:)` together with a `getStartedViewController(:, didFailWithError:)` without having to worry about name conflicts of a single `didFailWithError` delegate function.